test/xpu_api/utilities/utility/forward/forward.pass.cpp

```diff
@@ -0,0 +1,105 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/utility>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct A
+{
+};
+
+A
+source() TEST_NOEXCEPT
+{
+    return A();
+}
+const A
+csource() TEST_NOEXCEPT
+{
+    return A();
+}
+
+constexpr bool
+test_constexpr_forward()
+{
+    int x = 42;
+    const int cx = 101;
+    return dpl::forward<int&>(x) == 42 && dpl::forward<int>(x) == 42 && dpl::forward<const int&>(x) == 42 &&
+           dpl::forward<const int>(x) == 42 && dpl::forward<int&&>(x) == 42 && dpl::forward<const int&&>(x) == 42 &&
+           dpl::forward<const int&>(cx) == 101 && dpl::forward<const int>(cx) == 101;
+}
+
+void
+kernel_test()
+{
+    A a;
+    const A ca = A();
+
+    ((void)a);  // Prevent unused warning
+    ((void)ca); // Prevent unused warning
+
+    static_assert(dpl::is_same_v<decltype(dpl::forward<A&>(a)), A&>);
+    static_assert(dpl::is_same_v<decltype(dpl::forward<A>(a)), A&&>);
+    static_assert(dpl::is_same_v<decltype(dpl::forward<A>(source())), A&&>);
+    ASSERT_NOEXCEPT(dpl::forward<A&>(a));

```

Comment:
It seems like `ASSERT_NOEXCEPT` was useful before C++11 was supported. I propose to use `static_assert(noexcept(...))` directly.

Reply:
Fixed: always use `ASSERT_NOEXCEPT` and `ASSERT_NOT_NOEXCEPT` in all tests